libraries/Microsoft.Bot.Builder.Dialogs/SkillDialog.cs

```diff
@@ -119,13 +117,13 @@ private static SkillDialogArgs ValidateBeginDialogArgs(object options)
         private async Task SendToSkillAsync(ITurnContext context, Activity activity, CancellationToken cancellationToken)
         {
             // Create a conversationId to interact with the skill and send the activity
-            var skillConversationId = await _dialogOptions.ConversationIdFactory.CreateSkillConversationIdAsync(activity.GetConversationReference(), cancellationToken).ConfigureAwait(false);
+            var skillConversationId = await DialogOptions.ConversationIdFactory.CreateSkillConversationIdAsync(activity.GetConversationReference(), cancellationToken).ConfigureAwait(false);
 
             // Always save state before forwarding
             // (the dialog stack won't get updated with the skillDialog and things won't work if you don't)
-            var skillInfo = _dialogOptions.Skill;
-            await _conversationState.SaveChangesAsync(context, true, cancellationToken).ConfigureAwait(false);
-            var response = await _dialogOptions.SkillClient.PostActivityAsync(_dialogOptions.BotId, skillInfo.AppId, skillInfo.SkillEndpoint, _dialogOptions.SkillHostEndpoint, skillConversationId, activity, cancellationToken).ConfigureAwait(false);
+            var skillInfo = DialogOptions.Skill;
+            await DialogOptions.ConversationState.SaveChangesAsync(context, true, cancellationToken).ConfigureAwait(false);
+            var response = await DialogOptions.SkillClient.PostActivityAsync(DialogOptions.BotId, skillInfo.AppId, skillInfo.SkillEndpoint, DialogOptions.SkillHostEndpoint, skillConversationId, activity, cancellationToken).ConfigureAwait(false);
```

Comment:
If activity.DeliverMode equals DeliveryMode.BufferedReplies the skill will be complete for this turn and will have the responding activities here on this POST response. In which case, presumably this code should iterate over the list and send each activity. And then complete this Dialog.

Reply:
@gabog we will a separate issue to track this.